### ISSUE TITLE
Performance of fetch*** operations

### ISSUE THREAD

**Issue by ha62791** [cloudera/impyla#239]
Is there an performance improvement if you fetch the rows in below ways?
```
    cur.execute('SELECT * FROM data')
    fetchSize = 100
    rows = cur.fetchmany(fetchSize)
    while rows:
        '''
        do sth on rows
        '''
        rows = cur.fetchmany(fetchSize)
```
Does `fetchmany` fetch all the rows from Impala server on behind like fetchall ?
Also, what is the usage of fetchcolumnar ?

**Comment by ha62791** [comment]
From #11 and #221 ,
the default fetch batch size is 1024 so maybe using fetch*** is just for client-side optimization ?

**Comment by wesm** [comment]
If you care about fetch performance, you may also like to get involved with https://github.com/cloudera/hs2client -- I'm looking for contributors to help

**Comment by rizaon** [comment]
@csringhofer has #447 solved this?

**Comment by csringhofer** [comment]
@rizaon 
All fetch operations (fetchmany, fetchone, fetchall, fetchcolumnar ...) fetch from Impala server in batches of property "buffersize" - 10240 by default, can be changed with set_arraysize(). https://github.com/cloudera/impyla/pull/447 only changed the default from 1024.

Where fetchone/fetchmany/fetchall differ is how the results are converted from HS2's column oriented format to the row oriented results expected in DB api. The type conversion, e.g. for timestamps is already done at this point. Doing this transposition is more efficient in batches than one-by-one. https://github.com/cloudera/impyla/pull/444 improved the speed of these bulk operations significantly.

**Comment by csringhofer** [comment]
I wrote some comments about measurement of fetchmany/fetchcolumnar in https://github.com/cloudera/impyla/issues/543
TLDR: fetchcolumnar is significantly faster
I didn't experience much difference betwwen fetchall and fetchmany
